front/app/containers/LandingPage/SignUpButton.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import Button from 'components/UI/Button';
+import styled from 'styled-components';
+import { media } from 'utils/styleUtils';
+import { FormattedMessage } from 'utils/cl-intl';
+import messages from './messages';
+
+const CTAButton = styled(Button).attrs(() => ({
```

Comment:
Because I copied it from this piece of code

```tsx
        <SignUpButton
          fontWeight="500"
          padding="13px 22px"
          buttonStyle={buttonStyle}
          onClick={signUpIn}
          text={formatMessage(messages.createAccount)}
          className="e2e-signed-out-header-cta-button"
        />
 ```
 
 And this component has all these values as props. Not sure though if it's not the best way.

Reply:
The way you're using it now looks good. (I didn't know of the attrs method.)

With the Button component, we're passing a lot of styles as props indeed. It's a component that's probably doing too much though.